### DISCUSSION TITLE
Sub-daily budget failed

### DISCUSSION THREAD

**Discussion by utlacs** [EDmodel/ED2#398] [General]
Hello All,
I am running ED simulations for a region. Bit I am getting this error, "Sub-daily budget failed". Can anyone help me please. I have attached log file here. 
[logfile.txt](https://github.com/EDmodel/ED2/files/13559806/logfile.txt)

**Comment by mpaiao** [comment]
@utlacs I see that your log file has NaN, specifically in canopy air space CO2. Are you reading in CO2 from the meteorological drivers, or are you setting CO2 constant through ED2IN (Variable `NL%INITIAL_CO2`)? Note that if you have CO2 in the meteorological driver header (the file you specify in `NL%ED_MET_DRIVER_DB`, ED2 will use that variable instead of `NL%INITIAL_CO2`.

If both are set with reasonable numbers (units in either place should be in µmol mol<sup>&minus;1</sup>), then I suggest to recompile the model with full debugging settings so the model can stop at the first place an invalid number is used. There is a list of compilation flags useful for debugging with inter or gnu fortran that I left in this [Wiki page](https://github.com/EDmodel/ED2/wiki/Compiler-instructions-%28aka-the-include.mk-files%29) a couple of years ago. Some may have changed slightly with newer compiler versions, but they should be a good starting point.

Another quick test you can do before recompiling with debugger is to start your simulation during the Northern Hemisphere growing season (i.e., set `NL%IMONTHA=7`). It the simulation runs fine, it is likely a sign of a uninitialised variable when leaves of deciduous PFTs are off.

**Reply by utlacs** [reply]
Thank you for the reply. ED met driver has the problem. Upon resolving it. I am getting this error, "Likely to be a rejected step problem.".  Can you please look in to it.  I have attached my log file here:
[logfile.txt](https://github.com/EDmodel/ED2/files/13578234/logfile.txt). I have 12 files each year daily radiation data. My met driver file header is as follows: 1
/home/utlacs/projects/def-rkmishra-ab/utlacs/ED2-v.2.2.0/Input DATA/METDATA/ARF_
1 1 1.0 1.0 -66.420 45.970
12
 'vbdsf'   'vddsf'   'prate'  'dlwrf'   'pres'    'ugrd'     'vgrd'   'sh'     'tmp'     'hgt'    'nbdsf'   'nddsf'
 3600.0     3600.0   3600.0    3600.0   3600.0     3600.0    3600.0    3600.0   3600.0   3600.0    3600.0    3600.0
    0               0         1                 0         1              1              1            1         1             1             0         0

**Reply by xiangtaoxu** [reply]
I noticed your root_resp (fine root respiration) is NaN while GPP and leaf_respiration are all valid numbers. This NaN cascades into canopy airspace CO2 and resutl in the error.

root_resp is calculated from root biomass, a reference root respiration factor, and an Arrhenius function of temperature. If any one of them is NaN, root_resp would be NaN. Meanwhile, my guess is the temperature is wrong. It would be helpful to check your met forcing files and make sure all your variables are valid and have the right units. For instance, tmp should be in Kelvin not degC

**Reply by mpaiao** [reply]
@utlacs it seems the logfile.txt you attached in your second message (the one that starts with "Thank you for the reply") is the same as the first one, perhaps you attached the wrong file?

Regarding the met driver, I would strongly encourage to use flag "1" (time interpolation) for all variables except precipitation (which should be 0, no time interpolation).  Right now you are interpolating precipitation and not interpolating radiation. 

ED2 has a special time interpolator for radiation that accounts for the sun position, which makes the representation of the diurnal cycle much more realistic, as long as you set the variable `NL%IMETAVG` consistently with the data source (so you will need to find out this information for you met driver). `NL%IMETAVG` works as this:

1 — Averages ending at the time stamp (e.g. hourly data from 19 Jan 2019 2000 UTC are averages between 1900 and 2000 UTC).
2 — Averages starting at the time stamp (e.g. hourly data from 19 Jan 2019 2000 UTC are averages between 2000 and 2100 UTC).
3 — Averages starting at the time stamp (e.g. hourly data from 19 Jan 2019 2000 UTC are averages between 1930 and 2030 UTC).

**Reply by utlacs** [reply]
@mpaiao @xiangtaoxu Thank you for reply. I have 12 below stated variables as in the header of ED_MET_driver for a month JAN   and each variable has 31*24 columns, which is 744 data averaged for each day from MERRA-2 NASA Power, which is why I have used the below header:
home/utlacs/projects/def-rkmishra-ab/utlacs/ED2-v.2.2.0/Input DATA/METDATA/ARF_
1 1 1.0 1.0 -66.420 45.970
12
'vbdsf' 'vddsf' 'prate' 'dlwrf' 'pres' 'ugrd' 'vgrd' 'sh' 'tmp' 'hgt' 'nbdsf' 'nddsf'
3600.0 3600.0 3600.0 3600.0 3600.0 3600.0 3600.0 3600.0 3600.0 3600.0 3600.0 3600.0
0 0 1 0 1 1 1 1 1 1 0 0.
And regarding NL%IMETAVG If I set it to 1 or 2  My model is not running. However, when I set to -1 or 3. My model is running.  I haves set the other parameters as follows: NL%IMETTYPE    = 1
   NL%ISHUFFLE    = 0
   NL%METCYC1     = 2014
   NL%METCYCF     = 2021
   NL%IMETAVG     = 3
   NL%IMETRAD     = 0
   NL%INITIAL_CO2 = 410.
Please advise me If I configured wrongly. This time I am getting this error.: Note: The following floating-point exceptions are signalling: IEEE_INVALID_FLAG IEEE_DIVIDE_BY_ZERO IEEE_UNDERFLOW_FLAG IEEE_DENORMAL
STOP fatal_error
double free or corruption (!prev)

Program received signal SIGABRT: Process abort signal.

Backtrace for this error:
#0  0x153093a8d730 in ???
#1  0x153093a8c8d5 in ???
#2  0x1530936dd97f in ???
#3  0x1530936dd901 in ???
#4  0x1530936c756a in ???
#5  0x1530937200a7 in ???
#6  0x153093727449 in ???
#7  0x1530937292cb in ???
#8  0x153093f22af3 in ???
#9  0x153094041a96 in ???
#10  0x1530940423e9 in ???
#11  0x1530940426b3 in ???
#12  0x153093f6e658 in ???
#13  0x15309404219d in ???
#14  0x153093e2fc4b in ???
#15  0x1530936dffc6 in ???
#16  0x1530936e0179 in ???
#17  0x153093a8f924 in ???
#18  0x6365be in fatal_error_
	at /home/utlacs/XXXX/ED2-v.2.2.0/ED/build/bin-opt-E/fatal_error.F90:63
#19  0x69102c in __lapse_MOD_met_sanity_check
	at /home/utlacs/XXXXED2-v.2.2.0/ED/build/bin-opt-E/lapse.f90:578
#20  0x458f91 in __ed_met_driver_MOD_update_met_drivers
	at /home/utlacs/XXXX/ED2-v.2.2.0/ED/build/bin-opt-E/ed_met_driver.f90:2350
#21  0x463281 in ed_model_
	at /home/utlacs/XXXX/ED2-v.2.2.0/ED/build/bin-opt-E/ed_model.F90:547
#22  0x404c49 in MAIN__
	at /home/utlacs/XXXX/ED2-v.2.2.0/ED/build/bin-opt-E/edmain.F90:290
#23  0x403d8c in main
	at /home/utlacs/XXXX/utlacs/ED2-v.2.2.0/ED/build/bin-opt-E/edmain.F90:16
/localscratch/spool/slurmd/job23516250/slurm_script: line 18: 1290647 Aborted                 (core dumped) ./ed_2.2-opt -f ED2IN
[slurm-23516250.txt](https://github.com/EDmodel/ED2/files/13598227/slurm-23516250.txt)
